flink-core/src/main/java/org/apache/flink/util/WrappingProxyUtil.java

```diff
@@ -27,10 +27,16 @@ private WrappingProxyUtil() {
 		throw new AssertionError();
 	}
 
+	@SuppressWarnings("unchecked")
 	public static <T> T stripProxy(T object) {
-		while (object instanceof WrappingProxy) {
+
+		T previous = null;
+
+		while (object instanceof WrappingProxy && previous != object) {
```

Comment:
Guarding against infinite looping in case the bottom-most object is also a `WrappingProxy`?

Reply:
Exactly, somebody could come up with the stupid idea to return ``this`` as inner object. Just to be on the safe side.